### PR TITLE
Matcher modifiers

### PR DESCRIPTION
**Pull request by kejwmen**
Introduces idea described [here](https://github.com/coduo/php-matcher/pull/40) with two modifiers: `CaseInsensitive` and `IgnoreExtraKeys`.

#126 is included in this PR, I will rebase after it is merged.

### CODE REVIEW

**Comment by kejwmen** [comment]
Const visibility is probably not a feature we need, I will restore php 7.0 compat later.

File: .travis.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ cache:
 
 matrix:
   include:
-    - php: 7.0
+    - php: 7.1
```
Comment (norzechowicz) [inline .travis.yml]: please revert to 7.0

File: composer.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
         }
     ],
     "require": {
-        "php": ">=7.0.0",
+        "php": ">=7.1.0",
```
Comment (norzechowicz) [inline composer.json]: please revert to 7.0

**Comment by vitalyiegorov** [comment]
@kejwmen + for this feature, maybe you can revert php version asked by @norzechowicz to merge this PR?

**Comment by kejwmen** [comment]
@vitalyiegorov 
sure, I was busy because of GDPR, will try to finish this PR next week.

**Comment by kejwmen** [comment]
@norzechowicz @vitalyiegorov 

Restored php 7.0 compat, ScalarMatcher supports `CaseInsensitive` modifier.

Could you take a look at my code, maybe we should write some additional integration tests?

**Comment by kejwmen** [comment]
@norzechowicz could you take a look here?

**Comment by JarJak** [comment]
What I don't like with this approach is that if I want to allow extra keys only in specific place inside nested array, I can't do this with modifier, because modifiers work on whole pattern.

**Pull request closed without merging**